Refactor and animate puzzle

AmauryMi8 opened the pull request:
- Create puzzle service to decouple  PuzzleViewModel
- Use ctor to create new object
- Add animation to PopupLabel custom control

Commit:
Encapsulate TryResult creation

Use Ctor to create new tryResult instead of method

Commit:
Update PuzzleService.cs

Commit:
add animated custom control

Commit:
Update PopupLabel.cs

Commit:
Update Api access and Fine tune UI

After API is deployed, need to update address in the app.
Clean up and adjust some UI element too.

Commit:
Update PuzzlePage.xaml

add boxview in Collection view header to make space to display last item entirely

MartinOpde commented on ChallengeMk2/Controls/PopupLabel.cs:
```diff
@@ -0,0 +1,152 @@
+
+namespace ChallengeMk2.Controls
+{
+    public class PopupLabel : ContentView
```
Use xaml for your custom controls

AmauryMi8 replied:
I do not used Xaml because i do not know how to do animation in xaml. 
But i just find that free microsoft eBook, chapter 22 on Animations. I will read that and i will make a try with xaml ;)
But i am not sure how to do animation within xaml without use of event. I have to find out.

MartinOpde replied:
I meant use XAML to create the view.  Then you can give x:name to your view elements and perform your animations in the code-behind.

AmauryMi8 replied:
oki i see. I do that

AmauryMi8 replied:
Add new custom control that uses xaml to define view.

MartinOpde commented on ChallengeMk2/Controls/PopupLabel.cs:
```diff
@@ -0,0 +1,152 @@
+            {
+                { 0, 0.9, new Animation(v => Content.Scale = v, 0.1, 1.5) },
+                { 0.9, 1, new Animation(v => Content.Scale = v, 1.5, 1) },
+                //{ 0, 1, new Animation(v => Content.Rotation = v, 0, 360) }
```
Remove unused comment

Commit:
Change requested issue (not control in xaml yet)

Commit:
Add new custom control "AnimatedLabel"

- Uses xaml to create view
- Uses code behind to bindable properties and animations
- Change all "PopupLabel" to "AnimatedLabel" in code

MartinOpde commented on ChallengeMk2/Controls/AnimatedLabel.xaml:
```diff
@@ -0,0 +1,19 @@
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+             mc:Ignorable="d"
+             x:Class="ChallengeMk2.Controls.AnimatedLabel"
+			 x:Name="This">
```
what is this 'x:Name="this"' used for ?
If it is necessary, use an explicit name

AmauryMi8 replied:
Oops, it was an oversight from previous testing (when i try to define the binding context of the page to itself).
It should be deleted

MartinOpde commented on ChallengeMk2/Controls/PopupLabel.cs:
```diff
@@ -142,8 +142,7 @@ void AnimatePopup()
             var anim = new Animation
```
this file should probably be deleted.

Commit:
Resolve requested changes

- Remove PopupLabel.cs
- Remove x:name="This" from animatedLabel.xaml
